model/parameterSheet.cc

```diff
@@ -49,31 +50,28 @@ namespace
   };
 }
 
-void ParameterSheet::populateItemvector() {
-  itemvector.clear();	
-  minsky().canvas.selection.recursiveDo(&GroupItems::items,
+void ParameterSheet::populateItemVector() {
+  itemVector.clear();	
+  minsky().canvas.model->recursiveDo(&GroupItems::items,
                                         [&](Items&, Items::iterator i) {                                 
-                                          itemvector.emplace_back(*i);
+                                          itemVector.emplace_back(*i);
                                           return false;
                                         });   	
-}	
+}
 
 void ParameterSheet::draw(cairo_t* cairo) const
 {   
-  cairo_rectangle(cairo,0,0,m_width,m_height);
-  cairo_stroke_preserve(cairo);
-  cairo_clip(cairo);
-
   try
     {	
       		
-      //auto& value=ports[0]->getVariableValue();
-      if (!itemvector.empty())
-      //if (item)
-        if (auto v=itemvector[0]->variableCast())
-        //if (auto v=item->variableCast())
+      if (!itemVector.empty())
+      {
+	   float x0=0.0, y0=0.0;//+pango.height();	
+	   double w,h,h_prev,lh; 
+       for (size_t i=0; i<itemVector.size(); ++i)
```

Comment:
for (auto& i: itemVector) is neater, unless you need to index into the vector for some reason later.

Reply:
Thanks. I've fixed this.